### PR TITLE
Simplify and clean new/free X509_VERIFY_PARAM code

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
##### Checklist
- [x] CLA is signed
##### Description of change

Simplify and new/free X509_VERIFY_PARAM code  

Also applicable to 1.1.0

### CODE REVIEW

File: crypto/x509/x509_vpm.c [hunk 1]
```diff
@@ -114,15 +87,25 @@ X509_VERIFY_PARAM *X509_VERIFY_PARAM_new(void)
...
+     */
+    param->inh_flags = 0;
+    param->depth = -1;
+    param->auth_level = -1; /* -1 means unset, 0 is explicit */
     return param;
 }
 
 void X509_VERIFY_PARAM_free(X509_VERIFY_PARAM *param)
 {
     if (!param)
```
Comment (richsalz) [inline crypto/x509/x509_vpm.c]: NULL for pointer comparison

Comment (FdaSilvaYY) [inline crypto/x509/x509_vpm.c]: Fixed, and rebased !

**Comment by richsalz** [comment]
+1.  @mattcaswell you want to merge, or me?

**Comment by mattcaswell** [comment]
Merged.

**Pull request closed without merging**